gateway/src/apicast/policy/token_introspection/apicast-policy.json

```diff
@@ -21,6 +21,11 @@
         "description": "Client Secret for the Token Introspection Endpoint",
         "type": "string"
       },
+      "use_oidc_issuer_endpoint": {
```

Comment:
When this is `true` we don't need ClientID + Secret anymore, right?

I wonder if it makes sense to use dynamic schema like in https://github.com/3scale/apicast/pull/741.
Using either https://github.com/mozilla-services/react-jsonschema-form#property-dependencies or https://github.com/mozilla-services/react-jsonschema-form#schema-dependencies could allow us to have two login methods: clientid+secret or automatic using oidc_issuer_endpoint.

@davidor wdyt?

Reply:
> When this is true we don't need ClientID + Secret anymore, right?

Yes. We don't need to define `client_id` and `client_secret` in the schema.

Reply:
What I mean is having either client_id+client_secret or use_oidc_issuer_endpoint.


Try following schema:
```json
{
  "type": "object",
  "properties": {
    "auth_type": {
      "type": "string",
      "enum": [
        "oidc_issuer_endpoint",
        "client_id+client_secret"
      ],
      "default": "client_id+client_secret"
    }
  },
  "required": [
    "auth_type"
  ],
  "dependencies": {
    "auth_type": {
      "oneOf": [
        {},
        {
          "properties": {
            "auth_type": {
              "enum": [
                "client_id+client_secret"
              ]
            },
            "client_id": {
              "type": "string"
            },
            "client_secret": {
              "type": "string"
            }
          },
          "required": [
            "client_id",
            "client_secret"
          ]
        }
      ]
    }
  }
}
```

In https://mozilla-services.github.io/react-jsonschema-form/

That would allow user to choose which auth method to use and then fill the credentials.
It should also be extensible to add more properties in the future.

Reply:
Changed.